code/game/objects/effects/anomalies.dm

```diff
@@ -419,4 +421,79 @@
 			if(EXPLODE_LIGHT)
 				SSexplosions.lowturf += T
 
+/obj/effect/anomaly/delimber
+	name = "delimber anomaly"
+	icon_state = "delimber_anomaly"
+	aSignal = /obj/item/assembly/signaler/anomaly/delimber
+	immortal = TRUE
+	var/ticks = 0
+	var/releasedelay = 15
+	var/range = 5
+	///Lists for zones and bodyparts to swap and randomize
+	var/static/list/zones = list(BODY_ZONE_HEAD, BODY_ZONE_CHEST, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/static/list/chests
+	var/static/list/heads
+	var/static/list/l_arms
+	var/static/list/r_arms
+	var/static/list/l_legs
+	var/static/list/r_legs
+	var/static/list/organs
+
+/obj/effect/anomaly/delimber/Initialize(mapload, new_lifespan, drops_core)
+	. = ..()
+	if(!chests)
+		chests = subtypesof(/obj/item/bodypart/chest)
+	if(!heads)
+		heads = subtypesof(/obj/item/bodypart/head)
+	if(!l_arms)
+		l_arms = subtypesof(/obj/item/bodypart/l_arm)
+	if(!r_arms)
+		r_arms = subtypesof(/obj/item/bodypart/r_arm)
+	if(!l_legs)
+		l_legs = subtypesof(/obj/item/bodypart/l_leg)
+	if(!r_legs)
+		r_legs = subtypesof(/obj/item/bodypart/r_leg)
+	if(!organs)
+		organs = subtypesof(/obj/item/organ)
+
+/obj/effect/anomaly/delimber/anomalyEffect(delta_time)
+	. = ..()
+	ticks += delta_time
+	if(ticks < releasedelay)
+		return
+	ticks -= releasedelay
+
+	swap_parts(range)
+
+/obj/effect/anomaly/delimber/proc/swap_parts(swap_range)
+	for(var/mob/living/carbon/nearby in range(swap_range, src))
+		var/obj/item/clothing/suit/wear_suit = nearby.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+		var/obj/item/clothing/head/wear_helmet = nearby.get_item_by_slot(ITEM_SLOT_HEAD)
+		if(wear_suit?.armor[BIO] == 100 && wear_helmet?.armor[BIO] == 100)
```

Comment:
If you wanna check them directly for their bio armor values, you should use `getRating`.
(Also just suggesting that they're changed to `>=` instead of `==`, just for sanity reasons.)
```suggestion
		if(wear_suit?.armor.getRating(BIO) >= 100 && wear_helmet?.armor.getRating(BIO) >= 100)
```

BUT there's a slightly easier option which you can just run `getarmor(type = BIO)` on the mob itself to get how much total armor of a certain type the mob has 
```suggestion
		if(nearby.getarmor(type = BIO) >= 100)
```

You can even go one layer deeper and use `run_armor_check()` to include a message displayed if the mob's protected. 
```suggestion
		if(nearby.run_armor_check(type = BIO, absorb_text = "Your armor protects you from [src]!") >= 100)
```

Reply:
Ohh I like that last one, does it work on base carbons too or only on carbon/human types?

Reply:
run_armor_check is a `living` level proc, so it'd work here